models.py

```diff
@@ -91,58 +440,71 @@ def g_unet(in_ch, out_ch, nf, batch_size=1, is_binary=False, name='unet'):
     x = LeakyReLU(0.2)(conv9)
     # nf*8 x 1 x 1
 
-    dconv1 = Deconvolution(nf * 8, (batch_size, nf*8, 2, 2), k=2, s=1)(x)
+    dconv1 = Deconvolution(nf * 8,
+                           get_deconv_shape(batch_size, nf * 8, 2, 2),
+                           k=2, s=1)(x)
     dconv1 = BatchNorm()(dconv1)
     dconv1 = Dropout(0.5)(dconv1)
-    x = merge([dconv1, conv8], **merge_params)
+    try:
+        x = concatenate_layers([dconv1, conv8], **merge_params)
+    except ValueError:
```

Comment:
This was for debugging since the error messages from theano and tensorflow are pretty ugly for mismatched concatentations